Projects/Chess/Board.cs

```diff
@@ -0,0 +1,218 @@
+﻿namespace Chess;
+
+public class Board
+{
+	public static char[] Letters { get; } = {'A', 'B', 'C', 'D', 'E', 'F', 'G', 'H'};
+
+	public static Dictionary<char, int> BoardColumns { get; } = new()
+	{
+		{'A', 1}, {'B', 2}, {'C', 3}, {'D', 4}, {'E', 5}, {'F', 6}, {'G', 7}, {'H', 8}
+	};
+	
+	// position status for a piece on the board currently
+	public object[,] Cases { get; set; }
+	public IList<Piece> Pieces { get; set; }
+
+	private Board()
+	{
+		Cases = new object[8, 8];
+		Pieces = new List<Piece>();
+	}
+	
+	public static Board NewGame()
+	{
+		var board = new Board();
+
+		board.InitGame();
+
+		return board;
+	}
+	
+	
+    public static Board NewEmpty() => new Board();
+    public Piece GetPiece(char column, int row) => (Cases[row - 1, BoardColumns[column] - 1 ] as Piece)!;
+
+    public T SetWhite<T>( char column, int row ) where T : Piece
+    {
+        var piece = Activator.CreateInstance(typeof(T), PieceColor.White) as T;
+        Pieces.Add(piece);
+
+        SetPiece(piece, column, row);
+
+        return piece;
+    }
+
+    public T SetBlack<T>(char column, int row) where T : Piece
+    {
+        var piece = Activator.CreateInstance(typeof(T), PieceColor.Black) as T;
+        Pieces.Add(piece);
+
+        SetPiece(piece, column, row);
+
+        return piece;
+    }
+
+    public void InitGame()
+    {
+        foreach( var column in BoardColumns.Keys )
+        {
+
+            SetWhite<Pawn>(column, 2);
+            SetBlack<Pawn>(column, 7);
+        }
+
+        // Set our pieces for the start of the game
+        SetWhite<Rook>('A', 1);
+        SetWhite<Rook>('H', 1);
+        SetBlack<Rook>('A', 8);
+        SetBlack<Rook>('H', 8);
+
+        SetWhite<Knight>('B', 1);
+        SetWhite<Knight>('G', 1);
+        SetBlack<Knight>('B', 8);
+        SetBlack<Knight>('G', 8);
+
+        SetWhite<Bishop>('C', 1);
+        SetWhite<Bishop>('F', 1);
+        SetBlack<Bishop>('C', 8);
+        SetBlack<Bishop>('F', 8);
+
+        SetWhite<Queen>('D', 1);
+        SetBlack<Queen>('D', 8);
+
+        SetWhite<King>('E', 1);
+        SetBlack<King>('E', 8);
+    }
+
+    // try to do a movement of piece
+    public MoveOutcome MovePiece(
+	    char column, 
+	    int row, 
+	    char targetColumn, 
+	    int targetRow, 
+	    PieceColor? color = null )
+    {
+        var result = new MoveOutcome();
+    
+        if(BoardColumns[targetColumn] < 1 || BoardColumns[targetColumn] > 8 || targetRow is < 1 or > 8)
+        {
+            result.IsSuccess = false;
```

Comment:
Instead of having it like that, I recommend either using `Result<T, TError>` type, or using *nullable*:
```cs
public MoveOutcome? MovePiece
```
Then you can just return null instead of having this property